src/RpcClient.ts

```diff
@@ -646,81 +650,118 @@ export class RpcClient {
     // Rebuild the transaction with the final instructions
     if (isVersioned) {
       const v0Message = new TransactionMessage({
-        instructions: instructions,
-        payerKey: payerKey,
-        recentBlockhash: recentBlockhash,
+        instructions,
+        payerKey,
+        recentBlockhash,
       }).compileToV0Message(lookupTables);
 
       versionedTransaction = new VersionedTransaction(v0Message);
 
       const allSigners = feePayer ? [...signers, feePayer] : signers;
       versionedTransaction.sign(allSigners);
 
-      return { smartTransaction: versionedTransaction, lastValidBlockHeight };
-    } else {
-      legacyTransaction = new Transaction().add(...instructions);
-      legacyTransaction.recentBlockhash = recentBlockhash;
-      legacyTransaction.feePayer = payerKey;
-
-      for (const signer of signers) {
-        legacyTransaction.partialSign(signer);
-      }
+      return {
+        txBuff: versionedTransaction.serialize(),
+        blockhash,
+        minContextSlot,
+      };
+    }
+    legacyTransaction = new Transaction().add(...instructions);
+    legacyTransaction.recentBlockhash = recentBlockhash;
+    legacyTransaction.feePayer = payerKey;
 
-      if (feePayer) {
-        legacyTransaction.partialSign(feePayer);
-      }
+    for (const signer of signers) {
+      legacyTransaction.partialSign(signer);
+    }
 
-      return { smartTransaction: legacyTransaction, lastValidBlockHeight };
+    if (feePayer) {
+      legacyTransaction.partialSign(feePayer);
     }
+
+    return {
+      txBuff: legacyTransaction.serialize(),
```

Comment:
Same comment here — we should make the serialization process more explicit by having it occur before we return

Reply:
Also, should we be returning the serialized transaction? Would it be nicer for the sake of customizability if we return the built transaction instead of the built, serialized transaction?